### PR TITLE
Add a module chooser for deleted message vault on distributed server

### PR DESCRIPTION
**Pull request by Arsnael**
If the vault is disabled, it should not bind related classes and not create a bucket on s3 either.

### CODE REVIEW

**Review by quantranhong1999 — COMMENTED** [review]
We should backport this to postgres-app too IMO.

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedJamesConfiguration.java [hunk 1]
```diff
@@ -216,6 +225,16 @@ public DistributedJamesConfiguration build() {
                 }
             });
 
+            VaultConfiguration vaultConfiguration = this.vaultConfiguration.orElseGet(() -> {
+                try {
+                    return VaultConfiguration.from(propertiesProvider.getConfiguration("deletedMessageVault"));
+                } catch (FileNotFoundException e) {
+                    return VaultConfiguration.DEFAULT;
```
Comment (chibenwa) [inline tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedJamesConfiguration.java:232]: Please let's remain tmail behaviour: enable the vault by default

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java [hunk 2]
```diff
@@ -479,4 +480,20 @@ private static Module chooseDropListsModule(DistributedJamesConfiguration config
...
+        if (vaultConfiguration.isEnabled() && vaultConfiguration.isWorkQueueEnabled()) {
+            return Modules.combine(
+                new DistributedDeletedMessageVaultModule(),
+                new DeletedMessageVaultRoutesModule());
+        }
+        if (vaultConfiguration.isEnabled()) {
+            return Modules.combine(
+                new CassandraDeletedMessageVaultModule(),
+                new DeletedMessageVaultRoutesModule());
+        }
```
Comment (chibenwa) [inline tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java:494]: How about JMAP routes for the vault: does it makes sense to bind them when vault is disabled?

**Review by chibenwa — COMMENTED** [review]
Missing a test showing we can start without vault.

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java (hunk 2 above)
Comment (Arsnael) [reply tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedServer.java]: It makes it crash, just fixed.

File: tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedJamesConfiguration.java [hunk 3]
```diff
@@ -229,7 +229,7 @@ public DistributedJamesConfiguration build() {
                 try {
                     return VaultConfiguration.from(propertiesProvider.getConfiguration("deletedMessageVault"));
                 } catch (FileNotFoundException e) {
-                    return VaultConfiguration.DEFAULT;
+                    return VaultConfiguration.ENABLED_DEFAULT;
```
Comment (chibenwa) [inline tmail-backend/apps/distributed/src/main/java/com/linagora/tmail/james/app/DistributedJamesConfiguration.java:232]: Dont we need to update the doc to document this behaviour?

**Review by vttranlina — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Missing the test on the bucket not being created when I delete a message....

File: tmail-backend/apps/distributed/src/test/resources/eml/htmlMail.eml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+Delivered-To: mister@james.org
```
Comment (chibenwa) [inline tmail-backend/apps/distributed/src/test/resources/eml/htmlMail.eml:1]: Please shprten the EML and remove unecessary headers.

**Review by chibenwa — APPROVED** [review]
Otherwise no reserves.

Thanks for adding the test.

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**